src/app/ui/tabs/sql.rs

```diff
@@ -44,18 +44,21 @@ pub fn render_sql_editor(area: Rect, buf: &mut Buffer, app: &App) {
 }
 
 pub fn render_sql_results(area: Rect, buf: &mut Buffer, app: &App) {
-    let block = Block::default().title(" Results ").borders(Borders::ALL);
-    if let Some(q) = app.state.sql_tab.query() {
-        if let Some(r) = q.results() {
-            if let Some(s) = app.state.sql_tab.query_results_state() {
-                let stats = Span::from(format!(
-                    " {} rows in {}ms ",
-                    q.num_rows().unwrap_or(0),
-                    q.execution_time().as_millis()
-                ))
-                .fg(tailwind::WHITE);
-                let block = block.title_bottom(stats).fg(tailwind::ORANGE.c500);
-                let maybe_table = record_batches_to_table(r);
+    let block = Block::default()
+        .title(" Results ")
+        .borders(Borders::ALL)
+        .title(Title::from(" Page ").alignment(Alignment::Right));
+
+    let results = app.execution().results();
+    let locked = tokio::task::block_in_place(|| results.blocking_lock());
```

Comment:
It would be nice if we didn't need to do this while rendering, but without it the app panics.